src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -1779,4 +1804,119 @@ public Flux<EquipmentModificationInfos> createShuntCompensator(UUID networkUuid,
         return Mono.fromRunnable(() -> networkModificationRepository.moveModifications(groupUuid, modificationsToMove, before));
     }
+
+    private Mono<Void> assertLineSplitWithVoltageLevelInfosNotEmpty(LineSplitWithVoltageLevelInfos lineSplitWithVoltageLevelInfos) {
+        return lineSplitWithVoltageLevelInfos == null ? Mono.error(new NetworkModificationException(LINE_SPLIT_ERROR,
+            "Missing required attributes to split a line")) : Mono.empty();
+    }
+
+    private List<ModificationInfos> execSplitLineWithVoltageLevel(NetworkStoreListener listener,
+        LineSplitWithVoltageLevelInfos lineSplitWithVoltageLevelInfos,
+        ReporterModel reporter, Reporter subReporter) {
+
+        Network network = listener.getNetwork();
+        UUID networkUuid = listener.getNetworkUuid();
+        VoltageLevelCreationInfos mayNewVL = lineSplitWithVoltageLevelInfos.getMayNewVoltageLevelInfos();
+
+        List<ModificationInfos> inspectable = doAction(listener, () -> {
+            if (listener.isApplyModifications()) {
+                Line line = network.getLine(lineSplitWithVoltageLevelInfos.getLineToSplitId());
+                if (line == null) {
+                    throw new NetworkModificationException(LINE_NOT_FOUND, lineSplitWithVoltageLevelInfos.getLineToSplitId());
+                }
+
+                String voltageLeveId;
+                if (mayNewVL != null) {
+                    createVoltageLevelAction(mayNewVL, subReporter, network);
+                    voltageLeveId = mayNewVL.getEquipmentId();
+                } else {
+                    voltageLeveId = lineSplitWithVoltageLevelInfos.getExistingVoltageLevelId();
+                }
+
+                AttachVoltageLevelOnLine algo = new AttachVoltageLevelOnLine(
+                    lineSplitWithVoltageLevelInfos.getPercent(),
+                    voltageLeveId,
+                    lineSplitWithVoltageLevelInfos.getBbsOrBusId(),
+                    lineSplitWithVoltageLevelInfos.getNewLine1Id(),
+                    lineSplitWithVoltageLevelInfos.getNewLine1Name(),
+                    lineSplitWithVoltageLevelInfos.getNewLine2Id(),
+                    lineSplitWithVoltageLevelInfos.getNewLine2Name(),
+                    line);
+
+                algo.apply(network);
+
+                subReporter.report(Report.builder()
+                    .withKey("lineSplit")
+                    .withDefaultMessage("Line ${lineId} was split")
+                    .withValue("id", lineSplitWithVoltageLevelInfos.getLineToSplitId())
+                    .withSeverity(TypedValue.INFO_SEVERITY)
+                    .build());
+            }
+
+            listener.storeLineSplitWithVoltageLevelInfos(lineSplitWithVoltageLevelInfos);
+        }, LINE_SPLIT_ERROR, networkUuid, reporter, subReporter).stream().map(ModificationInfos.class::cast)
+            .collect(Collectors.toList());
+
+        if (!inspectable.isEmpty()) {
```

Comment:
Why this block ?
We always return one modification which represents the hypotheisis (an modification action) and not network modifications made by this action
And all impacted VLs are automatically added in the returned modification (action)

Reply:
I used it for the deletion of the line, because it does not fit with the "update only" that was sufficient with only one creation or one deletion, as we had before this hypothesis?